commons/com.b2international.index/src/com/b2international/index/mapping/DocumentMapping.java

```diff
@@ -157,21 +156,17 @@ public DocumentMapping(Class<?> type, DocumentMapping parent, boolean nested) {
 		}
 		checkRevisionType();
 				
-		this.nestedTypes = FluentIterable.from(getFields())
-			.transform(field -> {
+		this.nestedTypes = getFields().stream()
+			.map(field -> {
 				if (Reflections.isMapType(field)) {
 					return Map.class;
 				} else {
 					return Reflections.getType(field);
 				}
 			})
 			.filter(fieldType -> isNestedDoc(fieldType))
-			.toMap(new Function<Class<?>, DocumentMapping>() {
-				@Override
-				public DocumentMapping apply(Class<?> input) {
-					return new DocumentMapping(input, DocumentMapping.this.parent == null ? DocumentMapping.this : DocumentMapping.this.parent, true);
-				}
-			});
+			.distinct()
```

Comment:
Good catch! It looks like `FluentIterable#toMap` uses a `LinkedHashMap` internally while building the final (immutable) result, and so silently "de-duplicated" the keys. 👍